build-support/bin/generate_github_workflows.py

```diff
@@ -644,14 +679,15 @@ def build_steps(*, platform: Platform) -> list[Step]:
                         # virtualenv. This is because we must build both these things with
                         # multiple Python versions, whereas that caching assumes only one primary
                         # Python version (marked via matrix.strategy).
-                        *rust_caches(),
-                        *build_steps(platform=Platform.MACOS_X86_64),
+                        *helper_macos.rust_caches(),
+                        *build_steps(helper=helper_macos),
                         upload_log_artifacts(name="wheels-macos-x86_64"),
                         deploy_to_s3(),
                     ],
                 },
             }
         )
+    jobs.update(**macos_arm64_jobs())
```

Comment:
Rather than breaking out only the `macos_arm64_jobs`, it seems like there would be better symmetry to breaking out all of wheel building (i.e. lines 605-690) into a separate function `wheel_building_jobs`. As it stands, the parallels (and differences) between them aren't as obvious as they could be.

Reply:
Yeah, I have a TODO for ~that above.